python/cusignal/_spectral.py

```diff
@@ -270,14 +271,18 @@ def _get_backend_kernel(dtype, grid, block, stream, use_numba):
     )
 
 
-def _populate_kernel_cache():
-    for numba_type, c_type in _SUPPORTED_TYPES.items():
-        # JIT compile the numba kernels
-        sig = _numba_lombscargle_signature(numba_type)
-        _numba_kernel_cache[(str(numba_type))] = cuda.jit(sig, fastmath=True)(
-            _numba_lombscargle
-        )
+def _populate_kernel_cache(np_type, use_numba):
+
+    try:
+        numba_type, c_type = _SUPPORTED_TYPES.get(np_type)
+    except TypeError:
```

Comment:
The `TypeError` that this catches is `TypeError: cannot unpack non-iterable NoneType object` - whilst this code does "the right thing" in the end, it feels like a pretty weird pattern - I'd have expected using a getitem and catching a `KeyError` would be a little more straightforward to read and understand.

Reply:
@gmarkall Can you give me a hint of how I should do it differently?

Reply:
I've implemented `__getitem__` and `KeyError`